backend/dataall/modules/dataset_sharing/services/share_managers/lf_share_manager.py

```diff
@@ -417,3 +418,84 @@ def handle_revoke_failure(
             table, self.share, self.target_environment
         )
         return True
+
+    def handle_share_failure_for_all_tables(self, tables, error, share_item_status):
+        for table in tables:
+            share_item = ShareObjectRepository.find_sharable_item(
+                self.session, self.share.shareUri, table.tableUri
+            )
+            share_item_sm = ShareItemSM(share_item_status)
+            new_state = share_item_sm.run_transition(ShareObjectActions.Start.value)
+            share_item_sm.update_state_single_item(self.session, share_item, new_state)
+            new_state = share_item_sm.run_transition(ShareItemActions.Failure.value)
+            share_item_sm.update_state_single_item(self.session, share_item, new_state)
+
+            self.handle_revoke_failure(table=table, error=error)
+
+    def verify_catalog_ownership(self, catalog_dict):
+        if catalog_dict.get('account_id') != self.source_environment.AwsAccountId:
+            logger.info(f'Database {self.dataset.GlueDatabaseName} is a resource link and '
+                        f'the source database {catalog_dict.get("database_name")} belongs to a catalog account {catalog_dict.get("account_id")}')
+            if SessionHelper.is_assumable_pivot_role(catalog_dict.get('account_id')):
+                self.validate_catalog_ownership_tag(catalog_dict)
+            else:
+                raise Exception(f'Pivot role is not assumable, catalog account {catalog_dict.get("account_id")} is not onboarded')
+
+    def validate_catalog_ownership_tag(self, catalog_dict):
+        glue_client = GlueClient(account_id=catalog_dict.get('account_id'),
+                                 database=catalog_dict.get('database_name'),
+                                 region=catalog_dict.get('region'))
+
+        tags = glue_client.get_database_tags()
+        if tags.get('owner_account_id', '') == self.source_environment.AwsAccountId:
+            logger.info(f'owner_account_id tag exists and matches the source account id {self.source_environment.AwsAccountId}')
+        else:
+            raise Exception(f'owner_account_id tag does not exist or does not matches the source account id {self.source_environment.AwsAccountId}')
+
```

Comment:
Can we split `check_catalog_account_exists_and_update_processor` into 2 functions each with one task: `check_database_catalog_account` that gets the database source catalog and calls `_verify_catalog_ownership` and then depending on the output of this fucntion we can call `update_processor_source_account`

Reply:
Yes. I could do that.